spring.py

```diff
@@ -84,8 +84,8 @@ def initialize(self):
 
     def push(self, position, momentum, mass, spring_constant):
         p0 = momentum.copy()
-        momentum[:] = momentum - self.dt * spring_constant * position
         position[:] = position + self.dt * p0 / mass
+        momentum[:] = momentum - self.dt * spring_constant * position
```

Comment:
Your modification to this code has changed the algorithm from the "Forward Euler" method to the "Leapfrog" method. Instead of changing this class, you should just use the `Leapfrog` compute tool defined below.